src/utils/arrayUtils.ts

```diff
@@ -14,3 +14,14 @@ export function chunk<T>(array: T[], chunkSize: number): T[][] {
   // eslint-disable-next-line @typescript-eslint/no-unsafe-return
   return result
 }
+
+export async function callChunked<T>(
+  chunkSize: number,
+  array: readonly T[],
+  processFn: (arrayChunk: T[]) => Promise<void>,
+): Promise<void> {
+  for (let i = 0; i < array.length; i += chunkSize) {
+    const arrayChunk = array.slice(i, i + chunkSize)
```

Comment:
🔶 Very surprised that you don't use the `chunk` function directly above. If you did you could have:

```ts
const chunks = chunk(array, chunkSize)
for (const arrayChunk of chunks) {
  await processFn(arrayChunk)
} 
```

Reply:
This can also be done as a promise reducer:

```ts
chunk(array, chunkSize)
  .reduce(
    (promiseChain, chunk) => promiseChain.then(() => callback(chunk)),
    Promise.resolve()
  )
```

Reply:
And I'm not sure how I feel about using reduce here, sounds like an overcomplication. Original version is pretty simple.

Reply:
So, what should I do at the end?:)

Reply:
Personally I see no reason to change this code, for me it is as simple as it needs to be, I see no advantages, performance or complexity-wise in suggested alternatives.
@rupert-mckay Do you see this suggestion as RED blocking one?